tests/test_merge.py

```diff
@@ -26,3 +28,120 @@
 )
 def test_get_average_frame_count(samples: str, count: float) -> None:
     assert get_average_frame_count(samples) == count
+
+
+@pytest.mark.parametrize("insert_dso_name", [False, True])
+@pytest.mark.parametrize(
+    "stack, outcome_dict",
+    [
+        pytest.param(
+            "	 7f80 operator new+0x0 (/lib/libstdc++.so)",
+            dict(
+                dso_true="operator new (/lib/libstdc++.so)",
+                dso_false="operator new",
+            ),
+            id="operator_new",
+        ),
+        pytest.param(
+            "	 5501 [unknown] (/bin/cat)",
+            dict(
+                dso_true="(/bin/cat)",
+                dso_false="(/bin/cat)",
+            ),
+            id="unknown_bin_cat",
+        ),
+        pytest.param(
+            "	 7fd4 [unknown] (/lib/libudev.so (deleted))",
+            dict(
+                dso_true="(/lib/libudev.so (deleted))",
+                dso_false="(/lib/libudev.so (deleted))",
+            ),
+            id="libudev_deleted",
+        ),
+        pytest.param(
+            "	 1c00 [unknown] ([unknown])",
```

Comment:
On master, how does it look like? `[unknown]`?

Reply:
It was exactly this:
```
1c0830f0000 [unknown] ([unknown])
```

Reply:
I mean, how did the output of `_collapse_stack` look before your change, but anyway I verified now on master in IPython:
```
In [5]: _collapse_stack("aaa", " 1c00 [unknown] ([unknown])")                                                                                                                                              
Out[5]: 'aaa;[unknown]'
```

So looks good